### ISSUE TITLE
Sync not working

### ISSUE THREAD

**Issue by albsch** [nibdo/bloben-app#26]
I've successfully managed to add my CalDav account with two calendars in it. It seems the syncing does not work. I'm using the current `docker-compose` file.

Here is the error log:

```log
{"level":"error","message":"Unknown error: {\"stack\":\"TypeError: Cannot read properties of null (reading 'identity')\\n    at /usr/app/api/src/api/userEmailConfig/handlers/getUserEmailConfig.ts:34:33\\n    at Generator.next (<anonymous>)\\n    at fulfilled (/usr/app/api/build/api/userEmailConfig/handlers/getUserEmailConfig.js:5:58)\\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\",\"message\":\"Cannot read properties of null (reading 'identity')\"}","method":"GET","path":"/api/v1/users/email-config","tags":["UNKNOWN","REST"],"timestamp":"2022-05-16 10:00:39"}
{"level":"warn","message":"Too many requests for URL: /api/v1/sync METHOD: GET","tags":["REST","SECURITY"],"timestamp":"2022-05-16 10:08:56"}
{"level":"warn","message":"Too many requests for URL: /api/v1/sync METHOD: GET","tags":["REST","SECURITY"],"timestamp":"2022-05-16 10:09:05"}
{"level":"error","message":"Unknown error: {\"stack\":\"TypeError: Cannot read properties of null (reading 'identity')\\n    at /usr/app/api/src/api/userEmailConfig/handlers/getUserEmailConfig.ts:34:33\\n    at Generator.next (<anonymous>)\\n    at fulfilled (/usr/app/api/build/api/userEmailConfig/handlers/getUserEmailConfig.js:5:58)\\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\",\"message\":\"Cannot read properties of null (reading 'identity')\"}","method":"GET","path":"/api/v1/users/email-config","tags":["UNKNOWN","REST"],"timestamp":"2022-05-16 10:09:06"}
{"level":"warn","message":"Too many requests for URL: /api/v1/sync METHOD: GET","tags":["REST","SECURITY"],"timestamp":"2022-05-16 10:09:09"}
{"level":"error","message":"Unknown error: {\"stack\":\"TypeError: Cannot read properties of null (reading 'identity')\\n    at /usr/app/api/src/api/userEmailConfig/handlers/getUserEmailConfig.ts:34:33\\n    at Generator.next (<anonymous>)\\n    at fulfilled (/usr/app/api/build/api/userEmailConfig/handlers/getUserEmailConfig.js:5:58)\\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\",\"message\":\"Cannot read properties of null (reading 'identity')\"}","method":"GET","path":"/api/v1/users/email-config","tags":["UNKNOWN","REST"],"timestamp":"2022-05-16 10:09:09"}
```

**Comment by nibdo** [comment]
Hello, it might be this issue, if you have digest auth type.

https://github.com/nibdo/bloben-app/issues/23

Currently only basic is supported by used library.

**Comment by albsch** [comment]
I'm using `DavMail` as a proxy to a Microsoft Exchange server, to which I have no access to.

Instead I tried adding a CalDav calendar managed by `radicale`, but there seems to be another error occurring:

```
{"level":"info","message":null,"tags":["CRON"],"timestamp":"2022-05-16 15:10:00"}
{"level":"warn","message":"User login wrong password","tags":["REST","SECURITY"],"timestamp":"2022-05-16 16:21:07"}
{"level":"warn","message":"Too many requests for URL: /api/v1/users/login METHOD: POST","tags":["REST","SECURITY"],"timestamp":"2022-05-16 16:21:11"}
{"level":"error","message":"Unknown error: {\"stack\":\"TypeError: Cannot read properties of null (reading 'identity')\\n    at /usr/app/api/src/api/userEmailConfig/handlers/getUserEmailConfig.ts:34:33\\n    at Generator.next (<anonymous>)\\n    at fulfilled (/usr/app/api/build/api/userEmailConfig/handlers/getUserEmailConfig.js:5:58)\\n    at runMicrotasks (<anonymous>)\\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\",\"message\":\"Cannot read properties of null (reading 'identity')\"}","method":"GET","path":"/api/v1/users/email-config","tags":["UNKNOWN","REST"],"timestamp":"2022-05-16 16:21:18"}
{"level":"warn","message":"Too many requests for URL: /api/v1/caldav-calendars?component=VEVENT METHOD: GET","tags":["REST","SECURITY"],"timestamp":"2022-05-16 16:21:23"}
{"level":"warn","message":"Parsing event from caldav event string error [\"Invalid date: 20191208T160000Z,20191222T160000Z,20200112T160000Z,20200223T160000Z,\\r20200209T160000Z\"]","tags":["CRON","CALDAV"],"timestamp":"2022-05-16 16:22:06"}
{"level":"error","message":"Unknown error: {\"stack\":\"TypeError: Cannot read properties of null (reading '1')\\n    at /usr/app/api/node_modules/rrule/dist/es5/webpack:/rrule/src/parsestring.ts:133:27\\n    at Array.map (<anonymous>)\\n    at parseWeekday (/usr/app/api/node_modules/rrule/dist/es5/webpack:/rrule/src/parsestring.ts:125:15)\\n    at /usr/app/api/node_modules/rrule/dist/es5/webpack:/rrule/src/parsestring.ts:82:29\\n    at Array.forEach (<anonymous>)\\n    at parseRrule (/usr/app/api/node_modules/rrule/dist/es5/webpack:/rrule/src/parsestring.ts:56:9)\\n    at parseLine (/usr/app/api/node_modules/rrule/dist/es5/webpack:/rrule/src/parsestring.ts:42:14)\\n    at Array.map (<anonymous>)\\n    at Function.parseString (/usr/app/api/node_modules/rrule/dist/es5/webpack:/rrule/src/parsestring.ts:7:41)\\n    at Function.RRule.fromString (/usr/app/api/node_modules/rrule/dist/es5/webpack:/rrule/src/rrule.ts:114:28)\",\"message\":\"Cannot read properties of null (reading '1')\"}","method":"GET","path":"/api/v1/events","tags":["UNKNOWN","REST"],"timestamp":"2022-05-16 16:22:08"}
{"level":"warn","message":"Too many requests for URL: /api/v1/caldav-calendars?component=VEVENT METHOD: GET","tags":["REST","SECURITY"],"timestamp":"2022-05-16 16:22:08"}
{"level":"error","message":"Unknown error: {\"stack\":\"TypeError: Cannot read properties of null (reading '1')\\n    at /usr/app/api/node_modules/rrule/dist/es5/webpack:/rrule/src/parsestring.ts:133:27\\n    at Array.map (<anonymous>)\\n    at parseWeekday (/usr/app/api/node_modules/rrule/dist/es5/webpack:/rrule/src/parsestring.ts:125:15)\\n    at /usr/app/api/node_modules/rrule/dist/es5/webpack:/rrule/src/parsestring.ts:82:29\\n    at Array.forEach (<anonymous>)\\n    at parseRrule (/usr/app/api/node_modules/rrule/dist/es5/webpack:/rrule/src/parsestring.ts:56:9)\\n    at parseLine (/usr/app/api/node_modules/rrule/dist/es5/webpack:/rrule/src/parsestring.ts:42:14)\\n    at Array.map (<anonymous>)\\n    at Function.parseString (/usr/app/api/node_modules/rrule/dist/es5/webpack:/rrule/src/parsestring.ts:7:41)\\n    at Function.RRule.fromString (/usr/app/api/node_modules/rrule/dist/es5/webpack:/rrule/src/rrule.ts:114:28)\",\"message\":\"Cannot read properties of null (reading '1')\"}","method":"GET","path":"/api/v1/events","tags":["UNKNOWN","REST"],"timestamp":"2022-05-16 16:22:08"}
{"level":"error","message":"Unknown error: {\"stack\":\"TypeError: Cannot read properties of null (reading '1')\\n    at /usr/app/api/node_modules/rrule/dist/es5/webpack:/rrule/src/parsestring.ts:133:27\\n    at Array.map (<anonymous>)\\n    at parseWeekday (/usr/app/api/node_modules/rrule/dist/es5/webpack:/rrule/src/parsestring.ts:125:15)\\n    at /usr/app/api/node_modules/rrule/dist/es5/webpack:/rrule/src/parsestring.ts:82:29\\n    at Array.forEach (<anonymous>)\\n    at parseRrule (/usr/app/api/node_modules/rrule/dist/es5/webpack:/rrule/src/parsestring.ts:56:9)\\n    at parseLine (/usr/app/api/node_modules/rrule/dist/es5/webpack:/rrule/src/parsestring.ts:42:14)\\n    at Array.map (<anonymous>)\\n    at Function.parseString (/usr/app/api/node_modules/rrule/dist/es5/webpack:/rrule/src/parsestring.ts:7:41)\\n    at Function.RRule.fromString (/usr/app/api/node_modules/rrule/dist/es5/webpack:/rrule/src/rrule.ts:114:28)\",\"message\":\"Cannot read properties of null (reading '1')\"}","method":"GET","path":"/api/v1/events","tags":["UNKNOWN","REST"],"timestamp":"2022-05-16 16:22:09"}
{"level":"error","message":"Unknown error: {\"stack\":\"TypeError: Cannot read properties of null (reading '1')\\n    at /usr/app/api/node_modules/rrule/dist/es5/webpack:/rrule/src/parsestring.ts:133:27\\n    at Array.map (<anonymous>)\\n    at parseWeekday (/usr/app/api/node_modules/rrule/dist/es5/webpack:/rrule/src/parsestring.ts:125:15)\\n    at /usr/app/api/node_modules/rrule/dist/es5/webpack:/rrule/src/parsestring.ts:82:29\\n    at Array.forEach (<anonymous>)\\n    at parseRrule (/usr/app/api/node_modules/rrule/dist/es5/webpack:/rrule/src/parsestring.ts:56:9)\\n    at parseLine (/usr/app/api/node_modules/rrule/dist/es5/webpack:/rrule/src/parsestring.ts:42:14)\\n    at Array.map (<anonymous>)\\n    at Function.parseString (/usr/app/api/node_modules/rrule/dist/es5/webpack:/rrule/src/parsestring.ts:7:41)\\n    at Function.RRule.fromString (/usr/app/api/node_modules/rrule/dist/es5/webpack:/rrule/src/rrule.ts:114:28)\",\"message\":\"Cannot read properties of null (reading '1')\"}","method":"GET","path":"/api/v1/events","tags":["UNKNOWN","REST"],"timestamp":"2022-05-16 16:22:10"}
{"level":"error","message":"Unknown error: {\"stack\":\"TypeError: Cannot read properties of null (reading '1')\\n    at /usr/app/api/node_modules/rrule/dist/es5/webpack:/rrule/src/parsestring.ts:133:27\\n    at Array.map (<anonymous>)\\n    at parseWeekday (/usr/app/api/node_modules/rrule/dist/es5/webpack:/rrule/src/parsestring.ts:125:15)\\n    at /usr/app/api/node_modules/rrule/dist/es5/webpack:/rrule/src/parsestring.ts:82:29\\n    at Array.forEach (<anonymous>)\\n    at parseRrule (/usr/app/api/node_modules/rrule/dist/es5/webpack:/rrule/src/parsestring.ts:56:9)\\n    at parseLine (/usr/app/api/node_modules/rrule/dist/es5/webpack:/rrule/src/parsestring.ts:42:14)\\n    at Array.map (<anonymous>)\\n    at Function.parseString (/usr/app/api/node_modules/rrule/dist/es5/webpack:/rrule/src/parsestring.ts:7:41)\\n    at Function.RRule.fromString (/usr/app/api/node_modules/rrule/dist/es5/webpack:/rrule/src/rrule.ts:114:28)\",\"message\":\"Cannot read properties of null (reading '1')\"}","method":"GET","path":"/api/v1/events","tags":["UNKNOWN","REST"],"timestamp":"2022-05-16 16:22:10"}
{"level":"error","message":"Unknown error: {\"stack\":\"TypeError: Cannot read properties of null (reading '1')\\n    at /usr/app/api/node_modules/rrule/dist/es5/webpack:/rrule/src/parsestring.ts:133:27\\n    at Array.map (<anonymous>)\\n    at parseWeekday (/usr/app/api/node_modules/rrule/dist/es5/webpack:/rrule/src/parsestring.ts:125:15)\\n    at /usr/app/api/node_modules/rrule/dist/es5/webpack:/rrule/src/parsestring.ts:82:29\\n    at Array.forEach (<anonymous>)\\n    at parseRrule (/usr/app/api/node_modules/rrule/dist/es5/webpack:/rrule/src/parsestring.ts:56:9)\\n    at parseLine (/usr/app/api/node_modules/rrule/dist/es5/webpack:/rrule/src/parsestring.ts:42:14)\\n    at Array.map (<anonymous>)\\n    at Function.parseString (/usr/app/api/node_modules/rrule/dist/es5/webpack:/rrule/src/parsestring.ts:7:41)\\n    at Function.RRule.fromString (/usr/app/api/node_modules/rrule/dist/es5/webpack:/rrule/src/rrule.ts:114:28)\",\"message\":\"Cannot read properties of null (reading '1')\"}","method":"GET","path":"/api/v1/events","tags":["UNKNOWN","REST"],"timestamp":"2022-05-16 16:22:11"}
{"level":"error","message":"Unknown error: {\"stack\":\"TypeError: Cannot read properties of null (reading '1')\\n    at /usr/app/api/node_modules/rrule/dist/es5/webpack:/rrule/src/parsestring.ts:133:27\\n    at Array.map (<anonymous>)\\n    at parseWeekday (/usr/app/api/node_modules/rrule/dist/es5/webpack:/rrule/src/parsestring.ts:125:15)\\n    at /usr/app/api/node_modules/rrule/dist/es5/webpack:/rrule/src/parsestring.ts:82:29\\n    at Array.forEach (<anonymous>)\\n    at parseRrule (/usr/app/api/node_modules/rrule/dist/es5/webpack:/rrule/src/parsestring.ts:56:9)\\n    at parseLine (/usr/app/api/node_modules/rrule/dist/es5/webpack:/rrule/src/parsestring.ts:42:14)\\n    at Array.map (<anonymous>)\\n    at Function.parseString (/usr/app/api/node_modules/rrule/dist/es5/webpack:/rrule/src/parsestring.ts:7:41)\\n    at Function.RRule.fromString (/usr/app/api/node_modules/rrule/dist/es5/webpack:/rrule/src/rrule.ts:114:28)\",\"message\":\"Cannot read properties of null (reading '1')\"}","method":"GET","path":"/api/v1/events","tags":["UNKNOWN","REST"],"timestamp":"2022-05-16 16:22:12"}
```

**Comment by nibdo** [comment]
This looks now more like bug with date format. Can you please tell me, which bloben version are you using?

**Comment by albsch** [comment]
![image](https://user-images.githubusercontent.com/38429047/168757012-07394ddb-178f-4254-b88c-86579f84091c.png)

This is the version that is shown in the about page. I remembered that docker and api version was `0.8.0`, but for some reason it's blank now.

**Comment by nibdo** [comment]
Thanks, I will try to check, what could cause this.

**Comment by nibdo** [comment]
Unfortunately I won't be able do some quick fix here. :/ But I will try to investigate this more.

This line here:
`Parsing event from caldav event string error [\"Invalid date: 20191208T160000Z,20191222T160000Z,20200112T160000Z,20200223T160000Z,\\r20200209T160000Z\"]`

means that Bloben parsed iCal string wrong and tried to parse all those dates at once. I am quite curious what was original event here, because I had not seen something like this yet.

None of your events synced or only some are missing?

**Comment by albsch** [comment]
The calendar is completely empty.

**Comment by albsch** [comment]
I just checked the event, if this can help:
```
BEGIN:VEVENT
UID:longuid@google.com
DTSTART;TZID=Europe/Berlin:20191117T170000
DURATION:PT1H
CREATED:20191117T155634Z
DTSTAMP:20200619T102717Z
EXDATE:20191208T160000Z,20191222T160000Z,20200112T160000Z,20200223T160000Z,
 20200209T160000Z
RRULE:FREQ=WEEKLY;UNTIL=20200621;INTERVAL=1;BYDAY=SU
SEQUENCE:4
STATUS:CONFIRMED
SUMMARY:Practice Ballad
URL:https://www.google.com/calendar/event?eid=somelongid
X-MOZ-GENERATION:1
END:VEVENT
```

**Comment by nibdo** [comment]
New version 0.8.1 should handle parsing those dates better.

**Comment by albsch** [comment]
Still not working:

```
{"level":"error","message":"Unknown error: {\"stack\":\"TypeError: Cannot read properties of null (reading 'identity')\\n    at /usr/app/api/src/api/userEmailConfig/handlers/getUserEmailConfig.ts:34:33\\n    at Generator.next (<anonymous>)\\n    at fulfilled (/usr/app/api/build/api/userEmailConfig/handlers/getUserEmailConfig.js:5:58)\\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\",\"message\":\"Cannot read properties of null (reading 'identity')\"}","method":"GET","path":"/api/v1/users/email-config","tags":["UNKNOWN","REST"],"timestamp":"2022-05-23 08:07:40"}
{"level":"error","message":"Unknown error: {\"stack\":\"TypeError: Cannot read properties of null (reading '1')\\n    at /usr/app/api/node_modules/rrule/dist/es5/webpack:/rrule/src/parsestring.ts:133:27\\n    at Array.map (<anonymous>)\\n    at parseWeekday (/usr/app/api/node_modules/rrule/dist/es5/webpack:/rrule/src/parsestring.ts:125:15)\\n    at /usr/app/api/node_modules/rrule/dist/es5/webpack:/rrule/src/parsestring.ts:82:29\\n    at Array.forEach (<anonymous>)\\n    at parseRrule (/usr/app/api/node_modules/rrule/dist/es5/webpack:/rrule/src/parsestring.ts:56:9)\\n    at parseLine (/usr/app/api/node_modules/rrule/dist/es5/webpack:/rrule/src/parsestring.ts:42:14)\\n    at Array.map (<anonymous>)\\n    at Function.parseString (/usr/app/api/node_modules/rrule/dist/es5/webpack:/rrule/src/parsestring.ts:7:41)\\n    at Function.RRule.fromString (/usr/app/api/node_modules/rrule/dist/es5/webpack:/rrule/src/rrule.ts:114:28)\",\"message\":\"Cannot read properties of null (reading '1')\"}","method":"GET","path":"/api/v1/events","tags":["UNKNOWN","REST"],"timestamp":"2022-05-23 08:08:25"}
```

Using:
Docker image version 0.8.1
Calendar version 0.7.1
Api version 0.8.1

**Comment by nibdo** [comment]
Thank you. There were some line breaks in rows with dates, I hope next update (0.8.2) could fix this.

**Comment by albsch** [comment]
It works, thank you!

I have access to the raw events, if I knew which event was the cause I could look it up. I use DAVx⁵ clients to update the events. In the past I've used also various email clients (gmail, thunderbird, evolution, etc.).

I checked the event, since I redacted the <id>, maybe that was the culprit? It's actually split in two lines in the raw file for some reason:

![image](https://user-images.githubusercontent.com/38429047/170635118-536dc31a-cb49-4244-b848-967da34c345f.png)

Going to close this issue as the sync is working now.
